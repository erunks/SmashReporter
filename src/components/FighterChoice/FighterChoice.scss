@import '../../styles/base.scss';
$temp: white;

.FighterChoice {
  align-items: center;
  border: 4px solid $temp;
  display: flex;
  filter: drop-shadow(15px 15px 4px $temp);
  justify-content: center;
  margin: auto;
  position: relative;
  transform: skewY(-5deg);
  width: 100%;

  &.faceLeft {
    filter: drop-shadow(-15px 15px 4px $temp);
    transform: skewY(5deg);

    img {
      transform: scale(-1,1);
    }
  }

  &:hover {
    .helpText {
      @include visible(flex);
    }
  }

  .helpText {
    @include absolute-position;
    @include hidden;

    align-items: center;
    background-color: rgba(1, 1, 1, 0.5);
    color: white;
    font-variant: small-caps;
    height: 100%;
    margin: auto 0;
    text-align: center;
    width: 100%;
  }
}