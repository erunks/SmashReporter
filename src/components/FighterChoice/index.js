import React, { Fragment } from 'react';
import propTypes from 'prop-types';
import { buttonize } from 'helpers/jsx-a11y-helper'
import { URLS } from 'lib/imageAssets';

import './FighterChoice.scss';

const FighterChoice = ({ leftFacing, fighter, onClick }) => (
    <Fragment>
      <div
        className={`FighterChoice vertical ${leftFacing ? 'faceLeft' : ''}` }
        style={{ backgroundImage: `url(${URLS.fighterThumbnail(fighter)})` }}
        alt={`fighter: ${fighter}`} { ...buttonize(onClick) }
      >
        <div className="helpText center-container">
          <p>Click to Select Fighter</p>
        </div>
      </div>
      <div
        className={`FighterChoice horizontal ${leftFacing ? 'faceLeft' : ''}`}
        style={{ backgroundImage: `url(${URLS.fighterThumbnailIcon(fighter)})` }}
        alt={`fighter: ${fighter}`} {...buttonize(onClick)}
      >
        <div className="helpText center-container">
          <p>Click to Select Fighter</p>
        </div>
      </div>
    </Fragment>
);

const {
  bool,
  func,
  string
} = propTypes;

FighterChoice.propTypes = {
  leftFacing: bool,
  fighter: string.isRequired,
  onClick: func.isRequired
};

export default FighterChoice;
