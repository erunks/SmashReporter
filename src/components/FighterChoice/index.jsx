import React from 'react';
import propTypes from 'prop-types';
import { buttonize } from '../../helpers/jsx-a11y-helper'
import { URLS } from '../../lib/imageAssets';

import './FighterChoice.scss';

const FighterChoice = ({ leftFacing, fighter, onClick }) => (
    <div className={`FighterChoice ${leftFacing ? 'faceLeft' : ''}` } { ...buttonize(onClick) }>
    <img src={URLS.fighterThumbnail(fighter)} alt={`fighter: ${fighter}`} />
      <div className="helpText">
        <p>Click to Select Fighter</p>
      </div>
    </div>
);

const {
  bool,
  func,
  string
} = propTypes;

FighterChoice.propTypes = {
  leftFacing: bool,
  fighter: string.isRequired,
  onClick: func.isRequired
};

export default FighterChoice;
