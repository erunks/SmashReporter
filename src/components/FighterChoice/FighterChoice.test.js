import React from 'react';
import { shallow } from 'enzyme';
import { spy } from 'sinon';
import FighterChoice from '.';

describe('<FighterChoice>', () => {
  const props = {
    fighter: 'mario',
    onClick: spy()
  };
  const container = shallow(<FighterChoice leftFacing={false} {...props}/>);

  it('should match the snapshot', () => {
    expect(container.html()).toMatchSnapshot();
  });

  describe('when the fighter choice is clicked', () => {
    it('should call onClick', () => {
      const fighterChoice = container.find('.FighterChoice');
      fighterChoice.simulate('click');

      expect(props.onClick.called).toBe(true);
    });
  });

  describe('when leftFacing is true', () => {
    const container = shallow(<FighterChoice leftFacing={true} {...props} />);

    it('should add the class name appropriately', () => {
      const fighterChoice = container.find('.FighterChoice');

      expect(fighterChoice.hasClass('faceLeft')).toBe(true);
    })
  });
});
