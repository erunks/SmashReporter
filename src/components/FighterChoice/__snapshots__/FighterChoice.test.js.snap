// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FighterChoice> should match the snapshot 1`] = `"<div class=\\"FighterChoice vertical \\" style=\\"background-image:url(https://ik.imagekit.io/ozcerk4wii/fighter/thumb_v/mario.png)\\" alt=\\"fighter: mario\\" role=\\"button\\"><div class=\\"helpText center-container\\"><p>Click to Select Fighter</p></div></div><div class=\\"FighterChoice horizontal \\" style=\\"background-image:url(https://ik.imagekit.io/ozcerk4wii/fighter/thumb_a/mario.png)\\" alt=\\"fighter: mario\\" role=\\"button\\"><div class=\\"helpText center-container\\"><p>Click to Select Fighter</p></div></div>"`;
